File: basic_math/palindrome/palindrome.java
import java.util.*;
/*public class palindrome {
    public static void main(String[] args) {
        Scanner akil=new Scanner(System.in);
        String n=akil.next();

        String rev=new StringBuilder(n).reverse().toString();
        if(rev.equals(n)){
            System.out.print("palindrome");
        }
        else {
            System.out.print("not palindrome");
        }
        }
}
// another method

class palindrome{
    public static void main(String[] args) {
        Scanner akil=new Scanner(System.in);
        String n=akil.next();
        boolean is_palindrome=true;
        int left=0;
        int right=n.length()-1;
        while(left<right){
            if(n.charAt(left)!=n.charAt(right)){
                is_palindrome=false;
                break;
            }
            left++;
            right--;
        }
        if (is_palindrome){
            System.out.print("palindrome");
        }
        else{
            System.out.print("not palinfrome");
        }
    }
}*/

// another method
public class palindrome {

    public static void main(String[] args) {
        Scanner akil=new Scanner(System.in);
        String n=akil.next();
        String rev="";
        for(int i=0;i<n.length();i++){
            rev=n.charAt(i)+rev;
        }
        if(n.equals(rev)) System.out.print("palindrome");
        else System.out.print("not palindrome");
    }
}

File: basic_math/palindrome/palindrome.js
/*let n="121"
let rev=n.split('').reverse().join('');
if(n===rev){console.log("palindrome");}
else{console.log("not palindrome")}

//another methode

let n="121";
let left=0;
let right=n.length-1;
let ispalindrome=true;
while(left<right){
    if(n[left]!=n[right]){
        ispalindrome=false;
        break;
    }
    left++;
    right--;

}
if (ispalindrome){
    console.log("palinfrome");
}
else{
    console.log("not palindrome");
}*/

//another method

let n="121";
let rev="";
for(let i=0;i<n.length;i++){
    rev=n[i]+rev;
}
if(rev===n){
    console.log("palinfrome");
}
else{
    console.log("not a palindrome");
}